sinks/src/main/scala/com.snowplowanalytics.snowplow.eventgen/Http.scala

```diff
@@ -106,7 +106,11 @@ object Http {
         Stream
           .resource(httpClient)
           .flatMap(client => 
-            st.map(_._3).map(buildRequesst).evalMap(req => client.status(req)).void)
+            st
+            .map(_._3)
+            .map(_.getOrElse(throw new RuntimeException("Http sink received no data")))
```

Comment:
I dropped in a runtime exception because we to date have no sensible configuration handling of what formats go with what options - we just make assumptions about what the user wants to do. I'd like to change that, but until we do, this is just a janky stopgap - at least the app crashes when we configure the wrong format for the wrong target.

> We already have the logic for logging event count after it's done and completed.

I don't think we do - by my reading of the code (which isn't very reliable), for the other formats we are counting that _before_ passing to the sink. And it also appears not to be reliably accurate (but that's for another day). I couldn't see any logic in the app where the sink passes a result back up to the main flow of the app  - I think we _should_ do that for http target results at least - but it was beyond my abilities to get that done before I 'ran out of road' on this so to speak.